InnerEye/ML/utils/io_util.py

```diff
@@ -520,12 +516,21 @@ def load_images_from_dataset_source(dataset_source: PatientDatasetSource, check_
     # create raw sample to return
     metadata = copy(dataset_source.metadata)
     metadata.image_header = images[0].header
-    labels = load_labels_from_dataset_source(dataset_source, check_exclusive=check_exclusive, mask_size=mask.shape)
+    labels, missing_labels = load_labels_from_dataset_source(dataset_source, check_exclusive=check_exclusive)
+
+    # All class labels are counted, including missing or no missing ground truth class labels
+    assert len(missing_labels) == len(dataset_source.ground_truth_channels)
+    # Number of missing labels corresponds to the number of None objects in 'dataset_source.ground_truth_channels'
+    assert missing_labels.count(True) == dataset_source.ground_truth_channels.count(None)
+    # Number of not missing labels corresponds the number of provided ground truth (file name) channels.
+    assert missing_labels.count(False) == len(dataset_source.ground_truth_channels) - \
+           dataset_source.ground_truth_channels.count(None)
```

Comment:
Here and at other places: You tend to write a LOT of asserts. I would prefer to not litter the code with asserts that are as detailed as the ones here, in particular if all they do is check that the function just called does what the documentation says it should do. These detailed asserts are great for tests, but not in the main code.